### DISCUSSION TITLE
How to save trained model to a file and apply the model?

### DISCUSSION THREAD

**Discussion by zhangjingwei713** [opendilab/DI-engine#64] [Q&A] [answered]
I tested the test_smac_env.py and although it can run, the trained model is not saved. how can I save the model to a file, and how can I use the trained model like sl_model.pth for DI-star? Thanks!

**Comment by PaParaZz1** [comment, marked as answer]
- `test_smac_env.py` ([link](https://github.com/opendilab/DI-engine/blob/main/dizoo/smac/envs/test_smac_env.py)) is just a unittest file for smac environment without any training code.
- For SMAC training, you can enter `ding -m serial -c dizoo/smac/config/smac_3s5z_qmix_config.py -s 0` in your terminal, and model will be saved automatically in `default_experiment/ckpt` directory when better final reward is acquired.
- For more details of loading and saving model, you can refer to [doc](https://opendilab.github.io/DI-engine/quick_start/index.html#loading-saving-checkpoints).

**Reply by zhangjingwei713** [reply]
The model file `ckpt_best.pth.tar` is generated in the `default_experiment/ckpt` folder. Use `ding -m eval -c dizoo/smac/config/smac_2s3z_qmix_config.py -s 0 --load-path default_ experiment/ckpt/ckpt_best.pth.tar` command, I can see the battle between reinforcement-learning ai and the StarCratf 2 built-in ai, but the step speed is very fast like training. How can I set step speed to normal, and how to play the game with human vs agent mode? Thanks

**Reply by PaParaZz1** [reply]
Maybe you can try to enable this [argument](https://github.com/opendilab/DI-star/blob/main/ctools/pysc2/env/sc2_env.py#L134) in `SC2Env`, and find additional information in original pysc2 repo.

**Reply by upia99** [reply]
Add a time.sleep() [here](https://github.com/opendilab/DI-engine/blob/main/dizoo/smac/envs/smac_env.py#L440) if you want to slow down the game speed. Human vs agent is a bit of challenging. First, you need to be familiar with StartCraft2 editor, then you can edit the map which you trained your model with, replace the built-in AI with human player, then you can play with the model.
